core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1843,8 +1866,11 @@ class ReplicaManager(val config: KafkaConfig,
   }
 
   def markPartitionOffline(tp: TopicPartition): Unit = replicaStateChangeLock synchronized {
-    allPartitions.put(tp, HostedPartition.Offline)
-    Partition.removeMetrics(tp)
+    allPartitions.put(tp, HostedPartition.Offline) match {
+      case HostedPartition.Online(partition) =>
+        partition.fail()
```

Comment:
How about `markOffline` or something consistent with the name `markPartitionOffline`?

Reply:
`markOffline` sounds good to me.